CONTRIBUTING_A_NEW_QUEST.md

```diff
@@ -0,0 +1,396 @@
+
+Good places to find freely licensed images are [Geograph](https://www.geograph.org.uk/) and [Wikimedia Commons](https://commons.wikimedia.org/).
+
+You can also take your own photos, a standard smartphone camera is good enough. And sometimes a highly specific image is needed.
+
+In StreetComplete many images have unusual composition. Often it is necessary to leave space for a label at the bottom.
+
+Images should be free of visual debris, not misleading. Though it is fine to use an image not strictly matching what is depicted, as long as it is clear. For example, [a permanent pile of soil blocking road](https://wiki.openstreetmap.org/wiki/Tag:barrier%3Ddebris) is illustrated by a [temporary landslide](https://commons.wikimedia.org/wiki/File:Landslide_on_OR_42S_(46849629014).jpg). This is OK as images are illustrative.
+
+Photos go to a different folder than SVGs: they can be used directly by the build process so put them into folders
+  - [mdpi](app/src/main/res/drawable-mdpi) (typically 128 x 128 pixels, for rectangular ones smaller dimension should be 128 pixels)
+  - [hdpi](app/src/main/res/drawable-hdpi) (typically 192 x 192 pixels, for rectangular ones smaller dimension should be 192 pixels)
+  - [xhdpi](app/src/main/res/drawable-xhdpi) (typically 256 x 256 pixels, for rectangular ones smaller dimension should be 256 pixels)
+  - [xxhdpi](app/src/main/res/drawable-xxhdpi) (typically 384 x 384 pixels, for rectangular ones smaller dimension should be 384 pixels)
+
+Each of this folders should hold the same image resized to a different resolution. While testing various images it is enough to put into one folder.
+
+The [rescaling script](https://github.com/matkoniecz/rescaling_for_android) may be useful, but you can also do this manually with Gimp or similar software.
+
+After adding a file remember to update [the credit file](app/src/main/res/authors.txt) (different than one for icons).
+
+## Custom filters
+It is possible to use far more complex filters when querying for eligible elements.
+
+Matches like `surface ~ earth|dirt|ground` are possible and are evaluated as "`surface` is either of `earth`, `dirt`, `ground`"
+
+`access !~ private|no` will be evaluated to "`access` is neither `private` nor `no`"
+
+Simple lists are [evaluated as sets](https://github.com/streetcomplete/StreetComplete/blob/2e812b9a3b5288983309a7edde6e8f9db05ad3f2/app/src/test/java/de/westnordost/streetcomplete/data/elementfilter/filters/ElementFilterOverpassKtTest.kt#L79-L88) - and for example `surface ~ earth|dirt|ground` will not match `surface=earther` tag. Basically, `surface ~ earth|dirt|ground` is treated as `surface ~ ^earth|dirt|ground$`
+
+But using regexp like `surface ~ ^(.*)[0-9]$` is [also possible](https://github.com/streetcomplete/StreetComplete/blob/master/app/src/test/java/de/westnordost/streetcomplete/data/elementfilter/filters/ElementFilterOverpassKtTest.kt#L79-L88).
```

Comment:
Personally I'd not reference to the master branch in such a document, but pick a commit. This way, the line highlighting is still intact after years (in case this document get's outdated). This would also apply to the other links to code in this paragraph.

Reply:
That was one of things that was concerning to me. But I thought about not linking specific lines, but linking entire files as a replacement. This would have and additional benefit of links working locally an not being tied to GitHub.

If file gets changed significantly then stealthily linking outdated one is the worst case.

I guess that linking specific comment would help in case where content is up to date and just moved few lines up or down. I planned to monitor it and check after few months is it a big problem. I expect most or all linked code to be fairly stable.

Reply:
But if the file has been changed significantly, it's likely the documentation is wrong anyway (e.g. when the name of the commit message changed, or passing in new data like timestamps instead).

Reply:
it has _**already**_ proven to be unstable at several places: https://github.com/matkoniecz/Zazolc/issues/6, so I think linking to specific release tag is much better idea (as detailed at that issue).